lib/wallet/test/unit/Cardano/Write/Tx/BalanceSpec.hs

```diff
@@ -0,0 +1,1954 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{- HLINT ignore "Use null" -}
+{- HLINT ignore "Use camelCase" -}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- TODO: https://cardanofoundation.atlassian.net/browse/ADP-2841
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 902
+{-# OPTIONS_GHC -fno-warn-ambiguous-fields #-}
+#endif
+
+module Cardano.Write.Tx.BalanceSpec
+    ( spec
+
+    ----------------------------------------------------------------------------
+    -- TODO: ADP-3171
+    -- The exports in this section are currently needed by other tests related
+    -- to balanceTx that have not yet been relocated to this module. Once all
+    -- balanceTx-related tests have been relocated to this module, we should
+    -- remove these exports:
+    , dummyPolicyK
```

Comment:
`dummyPolicyK` shouldn't be needed for the sake of the `totalOutput` function where it's used. We can use the ledger functions instead.

Reply:
> `dummyPolicyK` shouldn't be needed for the sake of the `totalOutput` function where it's used. We can use the ledger functions instead.

Fair enough! Though in this PR I've purposefully avoided re-implementing anything. (To keep things simple.)

I look forward to us using more ledger functions in future PRs!